src/authentication/src/utils/onboarding.js

```diff
@@ -0,0 +1,124 @@
+import localforage from 'localforage'
+
+const ONBOARDING_SECRET_KEY = 'onboarding_secret'
+const ONBOARDING_STATE = 'onboarding_state'
+
+const generateRandomString = () => {
+  return Math.random()
+    .toString(36)
+    .substr(2, 11)
+}
+export const writeSecret = secret => {
+  return localforage.setItem(ONBOARDING_SECRET_KEY, secret)
+}
+
+export const readSecret = () => {
+  return localforage.getItem(ONBOARDING_SECRET_KEY)
+}
+
+export const clearSecret = () => {
+  return localforage.removeItem(ONBOARDING_SECRET_KEY)
+}
+
+export const writeState = state => {
+  return localforage.setItem(ONBOARDING_STATE, state)
+}
+
+export const readState = () => {
+  return localforage.getItem(ONBOARDING_STATE)
+}
+
+export const clearState = () => {
+  return localforage.removeItem(ONBOARDING_STATE)
+}
+const generateState = () => {
+  return generateRandomString()
+}
+const generateSecret = () => {
+  return generateRandomString()
+}
+export const checkIfOnboardingLogin = onboardingInformations => {
+  return onboardingInformations.code !== null
+}
+
+export const checkExchangedInformations = (
```

Comment:
Nit: I would personally inline this function, it does nothing complicated and putting it behind a function makes the thing a bit more opaque.